lib/src/model/canonicalization.dart

```diff
@@ -124,6 +124,16 @@ class ScoredCandidate implements Comparable<ScoredCandidate> {
   }
 
   @override
-  String toString() =>
-      "${library.name}: ${score.toStringAsPrecision(4)} - ${_reasons.join(', ')}";
+  String toString() {
```

Comment:
this is a pretty clear example of how the StringBuffer version is less readable.

Reply:
I wasn't going to change this one, but I thought it would be better to not include the ` - ` if _reasons is empty, so I decided the whole thing out to a StringBuffer since I was adding that logic anyway.

Reply:
I would strongly prefer the original style here -- I don't think `toString()` on ScoredCandidate is particularly performance impairing as scoring only occurs between libraries, there are a (relatively) small amount of libraries compared to other symbols, and even then we don't always use the scorer.